sale_commission/models/account_invoice.py

```diff
@@ -19,6 +19,20 @@ def _compute_commission_total(self):
         compute="_compute_commission_total",
         store=True,
     )
+    agent_ids = fields.Many2many(
+        "res.partner", string="Agents", compute="_compute_agents",
+        search="_search_agents")
+
+    @api.multi
+    def _compute_agents(self):
+        for so in self:
+            so.agent_ids = [(6, 0, so.mapped("invoice_line_ids.agents.agent").ids)]
+
+    @api.model
+    def _search_agents(self, operator, value):
+        sol_agents = self.env["sale.order.line.agent"].search(
+            [("agent", operator, value)])
+        return [('id', 'in', sol_agents.object_id.order_id.ids)]
```

Comment:
This is going to fail:
```suggestion
        return [('id', 'in', sol_agents.mapped("object_id.order_id").ids)]
```

Reply:
oops, fixed